goodreads_scraping/parsing/AuthorNamesParser.py

```diff
@@ -0,0 +1,31 @@
+from lxml import etree
+
+from .BaseParser import BaseParser
+
+
+class AuthorNamesParser(BaseParser):
+    def __init__(self, html):
+        self.xml = etree.fromstring(html)
+        self.parsed_author_names = self._parse_names()
+
+    @property
+    def get_xpath(self) -> str:
+        return "//div[@class='mainContentContainer']" + "//div[@class='rightContainer']" + "//div//h1//span"
+
+    def parse(self) -> tuple[str, str] | tuple[str, str, str]:
+        return (self.get_first_name(), self.get_middle_names(), self.get_last_name())
+
+    def _parse_names(self) -> str:
+        xpath_match = self.xml.xpath(self.get_xpath)[0]
+        text_lines = xpath_match.itertext()
+        return "\n".join(text_lines)
```

Comment:
1. Why do you use `itertext()` on `xpath_match`? It's useful if you want to extract all of the text from a tag that has different tags nested in it. Like this:
```
<div>
<p>All of the text</p>
<p>should be extracted</p>
<p>at once</p>
</div>
```
However, the XPATH that you've added leads to a simple `<span>` that has all of the text inside it without any nesting. In such case you can for example extract the `text` property from the `xpath_match`. Take a look at `goodreads_scraping/parsing/BookTitleParser.py` - it's done in there.

2. If you implement 1., you won't need to do any joins. However, it's still worth to mention that joining person names with a newline character is quite confusing. 😉 I guess you got inspired by a similar line in `goodreads_scraping/parsing/BookDescriptionParser.py`, but it's a different case. The text in there can be separated by multiple `<br>` tags, which we don't want to parse, but instead replace them with the newline characters. However, for the sake of parsing the author names, a simple space would be enough.